src/modules/session/thunks.ts

```diff
@@ -164,6 +167,32 @@ export const createAccount = createAsyncThunk(
   }
 );
 
+/**
+ * Delete a user account. Navigate to the login page.
+ */
+export const deleteAccount = createAsyncThunk(
+  "session/deleteAccount",
+  async (body: DeleteAccountRequest, { dispatch }) => {
+    const deleteAccountResponse = await dispatch(
+      sessionApi.endpoints.deleteAccount.initiate(body)
+    );
+
+    if ("error" in deleteAccountResponse) {
+      return;
+    }
+
+    dispatch(logOut());
```

Comment:
Doesn't completely remove state when using OAuth social accounts, after account deletion (no refresh or closing browser) the user is able to select their social and login again, old profile details persist

This is a strange edge case with cache/state where refresh or closing the browser fixes the issue

Note: When using the same OAuth login after delete account and refresh or close browser, user would get stuck in a loop of the create-profile steps, but again this is an edge case if the user decides to create another account with same socials right after deleting their first one. (username and password works normally in this case with no looping of create-profile)